### ISSUE TITLE
Hue icons not always showing

### ISSUE THREAD

**Issue by jeroen84s** [arallsopp/hass-hue-icons#206]
Hue icons is not always showing when I open the Home Assistant app, or when I open HA in the chrome / edge browser. Sometimes refreshing solves the problem, but not always. Original HA icons works always correctly. Re-installation is not working
![image](https://user-images.githubusercontent.com/44364045/198992504-71d50013-2cb0-47b5-976b-3d43ed682577.png)

**Comment by gnoffer** [comment]
i've the same issue !
![image](https://user-images.githubusercontent.com/7759889/199981922-e921b90d-6061-45cb-aa14-4fdf6d1ca886.png)

**Comment by m4d-maNu** [comment]
Same here, not always visible.

**Comment by jaredgudnason** [comment]
also having this issue currently.  Hue icons suddenly not showing / not listed in dropdown.
uninstalling and re-installing  component via HACS not fixing issue.

**Comment by RezzZ** [comment]
same, need to refresh the page or ios app one or more times for the icons to reappear. bug in HA? Can't imagine this is actually related to a file thats just available in HA.

**Comment by arallsopp** [comment]
Will keep an eye on this. Could be a memory / browser thing where the library (which always grows in size as icons are added) starts to crash some part of memory management, but suspect this is more likely HA than hass-hue-icons related. 

I've not seen it at all. Anyone have console output from a browser or client where they don't appear?

**Comment by RezzZ** [comment]
sorry for the n00b question but how do I create a console output out of chrome?
![image](https://user-images.githubusercontent.com/70499/214394339-aee8cdf6-b8ee-42c6-b22e-2a1321d3d001.png)

**Comment by arallsopp** [comment]
No worries. 

To open the developer console in Google Chrome, open the Chrome Menu in the upper-right-hand corner of the browser window and select More Tools > Developer Tools.

You can also use Option + ⌘ + J (on macOS), or Shift + CTRL + J (on Windows/Linux).

You want the JavaScript console which should tell you (on refresh) what libraries loaded and any errors it encounters.

**Comment by RezzZ** [comment]
on refresh I see this in my console:
![image](https://user-images.githubusercontent.com/70499/214396003-e6d4fe87-9b36-40b9-8664-a282d6d043b6.png)

**Comment by RezzZ** [comment]
the mentioned chrome extension is a popup blocker so i whitelisted my ha url now. This however is not possible on iOS and the missing images happens in the companion app as well

**Comment by arallsopp** [comment]
Ok. Did you follow this bit of the installation instructions?

Add the following to your configuration.yaml, save and restart HA.

```
frontend:
  extra_module_url:
    - /hacsfiles/hass-hue-icons/hass-hue-icons.js

```
Pointing it at whatever install location is used?

**Comment by RezzZ** [comment]
Hmm no I didn’t. Thought hacs would do all the magic for me 😳😅. I’ll get back to you and I will add the mentioned lines to my ha configuration yaml

**Comment by arallsopp** [comment]
Cheers. To be fair, HACS does a splendid job on the magic, but it can only load scripts after HA is done. That can lead to resources being unregistered on the first call to them.

**Comment by RezzZ** [comment]
Strange though that a page refresh solves it for a while..

**Comment by jeroen84s** [comment]
For me it's not solved with the installation instructions. I add this rule few months ago in my configuration.yaml

frontend:
  themes: !include_dir_merge_named themes
  extra_module_url:
    - /hacsfiles/hass-hue-icons/hass-hue-icons.js

**Comment by el97** [comment]
This happens mostly on the app for me. After a few restarts the icons shows up again.

**Comment by den-mac** [comment]
Yeah I was a happy user of these too for months but lately they just don't load for me. Tried everything mentioned here and it's not working for me anymore. I'll start replacing them with the usual mdi icons instead and not worry about it.

**Comment by arallsopp** [comment]
This seems to be an intermittent issue for a small subsection of users, and to do with the way HA handles front end libraries. 

I can't see an obvious fix for it from this side, other than to rework the entire library as an integration. 

That would let us install it with great privilege and control, but means reworking pretty much all of it. 

I'll think on this, as I don't want anyone to be without the icons they need.

**Comment by RezzZ** [comment]
I haven't had any issues since I installed it properly in the configuration.yaml

**Comment by arallsopp** [comment]
Thanks René, 

That's really good to know.

**Comment by Colorado4Wheeler** [comment]
I've noticed this same behavior.  I have several HACS icon sets installed and love the Hue icons, they really fit my need, but while they work fine on my desktop Chrome and Safari they don't work on my wall mounted tablets.  I have a Fire 10HD and they just won't draw at all, but all my other HACS plugins work.
